src/go/cmd/cr-syncer/syncer.go

```diff
@@ -295,6 +320,24 @@ func (s *crSyncer) processNextWorkItem(
 	}
 	defer q.Done(key)
 
+	// Restart informers on too many conflict errors
+	// client-go does not reliably recognize when watch calls are closed by remote API server
+	// cr-syncer is able to detect that when updating CRs on remote API server when there are multiple subsequent conflict errors (HTTP 409)
+	// like "...please apply your changes to the latest version and try again"
+	// This could occur at watchers of single CRDs while others keep working. Thus, it is less resource intensive just restarting informers of the affected CRDs rather than whoel cr-syncer
+	// Errors are counted in syncUpstream and syncDownstream functions
+	if s.conflictErrors >= 5 {
```

Comment:
Please make 5 a constant or ideally a flag, so that it is easy to play with the value by modifying the yaml. Using 5 as the default make sense.

Reply:
there is a flag called `conflict-error-limit` now